test/logger.spec.ts

```diff
@@ -0,0 +1,168 @@
+        defaultLogger.warn('warn');
+        expect(warnSpy).toHaveBeenCalled();
+      });
+
+      it('should not call console info', () => {
+        const defaultLogger = new DefaultLogger();
+        const infoSpy = jest.spyOn(global.console, 'info').mockImplementation();
+        defaultLogger.info();
+        expect(infoSpy).not.toHaveBeenCalled();
+      });
+
+      it('should not call console debug', () => {
+        const defaultLogger = new DefaultLogger();
+        const debugSpy = jest.spyOn(global.console, 'debug').mockImplementation();
+        defaultLogger.debug();
+        expect(debugSpy).not.toHaveBeenCalled();
+      });
+    });
+
+    describe('Safe Logger', () => {
+      it('should use the default logger because the custom logger is missing a function', () => {
+        const errorSpy = jest.spyOn(global.console, 'error').mockImplementation();
+        const safeLogger = new SafeLogger({} as any);
+
+        expect(errorSpy).toBeCalledWith(
+          expect.objectContaining({ message: 'The provided logger is missing the error method.' })
+        );
+        // Checking the private logger
+        expect(safeLogger['logger']).toBeInstanceOf(DefaultLogger);
+      });
+
+      it('should use the default logger when a custom logger throws', () => {
+        const innerLogger = {
+          error: jest.fn(() => {
+            throw new Error('Logger is invalid');
+          }),
+          warn: jest.fn(() => {
+            throw new Error('Logger is invalid');
+          }),
+          info: jest.fn(() => {
+            throw new Error('Logger is invalid');
+          }),
+          debug: jest.fn(() => {
+            throw new Error('Logger is invalid');
+          }),
+        };
+        const safeLogger = new SafeLogger(innerLogger);
+
+        expect(() => safeLogger.error()).not.toThrow();
+        expect(innerLogger.error).toHaveBeenCalled();
+
+        expect(() => safeLogger.warn()).not.toThrow();
+        expect(innerLogger.warn).toHaveBeenCalled();
+
+        expect(() => safeLogger.info()).not.toThrow();
+        expect(innerLogger.info).toHaveBeenCalled();
+
+        expect(() => safeLogger.debug()).not.toThrow();
+        expect(innerLogger.debug).toHaveBeenCalled();
+      });
+
+      it('should create a safe logger', () => {
+        OpenFeature.logger = {} as any;
+        expect(OpenFeature.logger).toBeInstanceOf(SafeLogger);
+      });
+    });
+
+    it('should be an instance of the default logger', () => {
+      const client = OpenFeature.getClient();
+
+      expect(client.logger).toBeInstanceOf(DefaultLogger);
+    });
+
+    it('should provide a logger to the provider', async () => {
+      const client = OpenFeature.getClient();
+      const mockedLogger = new MockedLogger();
+      client.logger = mockedLogger;
+      await client.getBooleanValue('test', false);
+      expect(mockedLogger.info).toHaveBeenCalledWith('resolving boolean value');
```

Comment:
I think a lot of duplicated strings in assertions could be constants.

Reply:
Done